.github/workflows/main.yml

```
@@ -1,21 +1,65 @@
-on: [push]
+name: Tests
+on:
+  pull_request:
+  push:
+    branches:
+      - main
 
 jobs:
-  new_push_job:
+  unit_tests:
     runs-on: ubuntu-latest
-    name: New push to repo
     steps:
-    - name: Publish to slack
+    - uses: actions/checkout@v2
+    - run: npm ci && npm run build
+    - run: npm test
+
+  integration_test_botToken:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: npm ci && npm run build
+    - name: Post message to Slack via botToken
       id: slack
-      uses: slackapi/slack-github-action@v1.15.0
+      uses: ./
       with:
         channel-id: 'CTAAHRA79'
```

We may want to use a secret value for this too:
```suggestion
        channel-id: secrets.SLACK_CHANNEL_ID
```